api/src/testIntegration/java/bisq/api/http/UserEndpointIT.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.api.http;
+
+import bisq.api.http.model.ChangePassword;
+
+import bisq.common.util.Base64;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.isEmptyString;
+
+
+
+import com.github.javafaker.Faker;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import org.arquillian.cube.docker.impl.client.containerobject.dsl.Container;
+import org.arquillian.cube.docker.impl.client.containerobject.dsl.DockerContainer;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.junit.InSequence;
+
+@RunWith(Arquillian.class)
+public class UserEndpointIT {
+
+    private static String validPassword = new Faker().internet().password();
+    private static String invalidPassword = getRandomPasswordDifferentThan(validPassword);
+    @DockerContainer
+    Container alice = ContainerFactory.createApiContainer("alice", "8081->8080", 3333, false, false);
+
+    private static String getRandomPasswordDifferentThan(String otherPassword) {
+        String newPassword;
+        do {
+            newPassword = new Faker().internet().password();
+        } while (otherPassword.equals(newPassword));
+        return newPassword;
+    }
+
+    @InSequence
+    @Test
+    public void waitForAllServicesToBeReady() throws InterruptedException {
+        ApiTestHelper.waitForAllServicesToBeReady();
+        verifyThatAuthenticationIsDisabled();
+    }
+
+    @InSequence(2)
+    @Test
+    public void changePassword_missingPayload_returns400() {
+        int alicePort = getAlicePort();
+        given().
+                port(alicePort).
+                contentType(ContentType.JSON).
+                accept(ContentType.JSON).
+//
+        when().
+                post("/api/v1/user/password").
+//
+        then().
+                statusCode(400).
+                and().body(isEmptyString());
+
+        verifyThatAuthenticationIsDisabled();
+    }
+
+    @InSequence(3)
+    @Test
+    public void changePassword_settingFirstPassword_enablesAuthentication() {
```

Comment:
Does it mean that before someone set password he has access to every endpoint ? Is it intended ?

Reply:
Yes. Setting default password does not provide any security.
We could consider in future preventing access without password, but that would require some way to generate password file on the system where Bisq runs and somehow transferring it to the client, but that's out of scope for now.